File: yugabyte-integration/src/main/java/com/ondc/yugabyte_integration/Entity/PayloadDetailsDTO.java
package com.ondc.yugabyte_integration.Entity;

import java.util.List;

public class PayloadDetailsDTO {
    private SessionDetails.Type npType;
    private String domain;
    private Payload payloads;

    public PayloadDetailsDTO(SessionDetails.Type npType, String domain, Payload payloads) {
        this.npType = npType;
        this.domain = domain;
        this.payloads = payloads;
    }

    public SessionDetails.Type getNpType() {
        return npType;
    }

    public void setNpType(SessionDetails.Type npType) {
        this.npType = npType;
    }

    public String getDomain() {
        return domain;
    }

    public void setDomain(String domain) {
        this.domain = domain;
    }

    public Payload getPayloads() {
        return payloads;
    }

    public void setPayloads(Payload payloads) {
        this.payloads = payloads;
    }
}

File: src/entity/PayloadDetailsDTO.ts
import { SessionType, Type } from "./ActionEnums";
import { Payload } from "./Payload";

export class PayloadDetailsDTO {
  npType: string;
  domain: string;
  payload: InstanceType<typeof Payload>; // Keep Payload as type

  constructor(npType: string, domain: string, payload: InstanceType<typeof Payload> | any) {
    this.npType = npType;
    this.domain = domain;
    this.payload = payload;
  }

  getNpType(): string {
    return this.npType;
  }

  setNpType(npType: Type): void {
    this.npType = npType;
  }

  getDomain(): string {
    return this.domain;
  }

  setDomain(domain: string): void {
    this.domain = domain;
  }

  getPayload(): InstanceType<typeof Payload> {
    return this.payload;
  }

  setPayload(payload: InstanceType<typeof Payload> | any): void {
    this.payload = payload;
  }
}
